libsolidity/codegen/ir/IRGenerator.cpp

```diff
@@ -532,27 +547,39 @@ std::string IRGenerator::generateGetter(VariableDeclaration const& _varDecl)
 		{
 			solAssert(paramTypes.empty(), "");
 			solUnimplementedAssert(type->sizeOnStack() == 1);
-			return Whiskers(R"(
+
+			auto t = Whiskers(R"(
 				<astIDComment><sourceLocationComment>
 				function <functionName>() -> rval {
-					rval := loadimmutable("<id>")
+					<?eof>
+						rval := auxdataloadn(<immutableDataSectionOffset>)
+					<!eof>
+						rval := loadimmutable("<id>")
+					</eof>
 				}
 				<contractSourceLocationComment>
-			)")
-			(
+				)");
+			t(
 				"astIDComment",
 				m_context.debugInfoSelection().astID ?
 					"/// @ast-id " + std::to_string(_varDecl.id()) + "\n" :
 					""
-			)
-			("sourceLocationComment", dispenseLocationComment(_varDecl))
-			(
+			);
+			t("sourceLocationComment", dispenseLocationComment(_varDecl));
+			t(
 				"contractSourceLocationComment",
 				dispenseLocationComment(m_context.mostDerivedContract())
-			)
-			("functionName", functionName)
-			("id", std::to_string(_varDecl.id()))
-			.render();
+			);
+			t("functionName", functionName);
+
+			auto const eof = m_context.eofVersion().has_value();
+			t("eof", eof);
+			if (!eof)
+				t("id", std::to_string(_varDecl.id()));
+			else
+				t("immutableDataSectionOffset", std::to_string(m_context.immutableMemoryOffset(_varDecl)));
```

Comment:
This is not an offset relative to the start of data section but rather aux data. I'd rename it to avoid misunderstandings:
```suggestion
				t("immutableAuxDataOffset", std::to_string(m_context.immutableMemoryOffset(_varDecl)));
```

Reply:
Also, note that `immutableMemoryOffset()` gives you an absolute memory address. I'm not sure it's suitable for use as data section offset. It needs to be made relative and also, I'm not sure these offsets are contiguous.

Reply:
Ohh shit. You are right. It' relative. I will change accordingly.